### PR TITLE
Complete serializer for IonQ gates

### PR DESCRIPTION
**Pull request by dabacon**

### CODE REVIEW

**Comment by dabacon** [comment]
This could probably also use `functools.singledispatch`, but my guess is that this is less known and so less readable for most. Sounds like something @viathor would have an opinion on :)

**Comment by dabacon** [comment]
Part of #3479

**Comment by dabacon** [comment]
@wcourtney for Google owners.

**Review by wcourtney — APPROVED** [review]
Thanks for updating our comments for clarity!

File: cirq/ionq/serializer.py [hunk 1]
```diff
@@ -30,6 +31,27 @@ class Serializer:
...
+    def __init__(self, atol: float = 1e-8):
+        """Create the Serializer.
+
+        Args:
+            atol: Absolute tolerance used in determining whether a gate with
+                a float parameter should be serialized as a gate rounded
+                to that parameter. Defaults to 1e-8.
+        """
+        self.atol = atol
+        self.dispatch: Dict[Type['cirq.Gate'], Callable] = {
```
[the remark this answers is not in the document]
Comment (dabacon) [reply cirq/ionq/serializer.py]: done

**Review by nakardo — APPROVED** [review]
LGTM!

**Pull request merged**